src/components/AddStudent.js

```diff
@@ -0,0 +1,95 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
+import Cookies from 'js-cookie';
+import {SERVER_URL} from '../constants.js'
+
+class AddStudent extends Component{
+    constructor(props){
+        super(props);
+        this.state = {open: false, name: '', email: ''};
+    };
+
+    handleClickOpen = () => {
+        this.setState({open:true});
+    };
+
+    handleClose = () => {
+        this.setState({open:false});
+    };
+
+    handleChangeName = (event) => {
+        this.setState({name: event.target.value});
+    };
+
+    handleChangeEmail = (event) => {
+        this.setState({email: event.target.value});
+    };
+
+    handleAdd = () => {
+        //console.log(this.state.name);
+        //console.log(this.state.email);
+        
+        const token = Cookies.get('XSRF-TOKEN');
+        fetch(`http://localhost:8080/student/?email=${this.state.email}&name=${this.state.name}`,
+        {
+            method: 'POST',
+            headers: {'X-XSRF-TOKEN': token, 'Content-Type': 'application/json'},
+            //I did have to enable CORS in the backend springboot app for this to work
+        }).then((response) => {
+            console.log(response.json());
+            if(response.status == 200){
+                toast.success("Student added successfully.", {
+                    position: toast.POSITION.BOTTOM_LEFT
+                });
+            } else if(response.status == 400){
+                toast.error("Student already exists.", {
+                    position: toast.POSITION.BOTTOM_LEFT
+                });
+            } else {
+                toast.error("Student could not be added.", {
+                    position: toast.POSITION.BOTTOM_LEFT
+                });
+            }
+        })
+        
```

Comment:
Maybe some more comments but looks good!

Reply:
will add comments and push directly to main